donjo-app/controllers/Penduduk.php

```diff
@@ -532,72 +575,70 @@ public function statistik($tipe = 0, $nomor = 0, $sex = NULL)
-			case 15: $_SESSION['umurx'] = $nomor;  $pre = "KATEGORI UMUR : "; break;
-			case 16: $_SESSION['cara_kb_id'] = $nomor; $pre = "CARA KB : "; break;
-			case 17:
-				$_SESSION['akta_kelahiran'] = $nomor;
-				if ($nomor <> BELUM_MENGISI) $_SESSION['umurx'] = $nomor;
-				$pre = "AKTA KELAHIRAN : ";
-				break;
+			case 0: $session = 'pendidikan_kk_id'; $kategori = 'PENDIDIKAN DALAM KK : '; break;
+			case 1: $session = 'pekerjaan_id'; $kategori = 'PEKERJAAN : '; break;
+			case 2: $session = 'status'; $kategori = 'STATUS PERKAWINAN : '; break;
+			case 3: $session = 'agama'; $kategori = 'AGAMA : '; break;
+			case 4: $session = 'sex'; $kategori = 'JENIS KELAMIN : '; break;
+			case 5: $session = 'warganegara'; $kategori = 'WARGANEGARA : '; break;
+			case 6: $session = 'status_penduduk'; $kategori = 'STATUS PENDUDUK : '; break;
+			case 7: $session = 'golongan_darah'; $kategori = 'GOLONGAN DARAH : '; break;
+			case 9: $session = 'cacat'; $kategori = 'CACAT : '; break;
+			case 10: $session = 'menahun'; $kategori = 'SAKIT MENAHUN : '; break;
+			case 13: $session = 'umurx'; $kategori = 'UMUR (RENTANG) : '; break;
+			case 14: $session = 'pendidikan_sedang_id'; $kategori = 'PENDIDIKAN SEDANG DITEMPUH : '; break;
+			case 15: $session = 'umurx'; $kategori = 'UMUR (KATEGORI) : '; break;
+			case 16: $session = 'cara_kb_id'; $kategori = 'CARA KB : '; break;
+			case 17: $session = 'akta_kelahiran'; $kategori = 'AKTA KELAHIRAN : UMUR '; break;
+			case 19: $session = 'id_asuransi'; $kategori = 'JENIS ASURANSI : '; break;
+			case 'covid': $session = 'status_covid'; $kategori = 'STATUS COVID : '; break;
+			case 'bantuan_penduduk': $session = 'penerima_bantuan'; $kategori = 'PENERIMA BANTUAN PENDUDUK : '; break;
 			case 18:
 				if ($sex == NULL)
 				{
-					$_SESSION['status_ktp'] = 0;
-					$_SESSION['sex'] = ($nomor == 0) ? NULL : $nomor;
-					$sex = $_SESSION['sex'];
+					$this->session->status_ktp = 0;
+					$this->session->sex = ($nomor == 0) ? NULL : $nomor;
+					$sex = $this->session->sex;
 					unset($nomor);
 				}
 				else
-					$_SESSION['status_ktp'] = $nomor;
-				$pre = "KEPEMILIKAN WAJIB KTP : ";
-				break;
-			case 19:
-				$_SESSION['id_asuransi'] = $nomor; $pre = "JENIS ASURANSI : ";
-				break;
-			case 'covid':
-				$_SESSION['status_covid'] = $nomor; $pre = "STATUS COVID : ";
-				break;
-			case 'bantuan_penduduk':
-				$_SESSION['penerima_bantuan'] = $nomor; $pre = "PENERIMA BANTUAN (PENDUDUK) : ";
+				{
+					$this->session->status_ktp = $nomor;
+				}
+
+				$kategori = "KEPEMILIKAN WAJIB KTP";
 				break;
 		}
+
+		// Filter berdasarkan kategori tdk dilakukan jika $nomer = TOTAL (888)
+		if ($tipe != 18 && $nomor != TOTAL)
+		{
+			$this->session->$session = $nomor;
+		}
+
 		$judul = $this->penduduk_model->get_judul_statistik($tipe, $nomor, $sex);
 		// Laporan wajib KTP berbeda - menampilkan sebagian dari penduduk, jadi selalu perlu judul
-		if ($judul['nama'] or $tipe = 18)
+		if ($judul['nama'] OR $tipe = 18)
```

Comment:
Coba lihat apa style guide CI untuk penulisan ini. Dan masukkan ke style guide OpenSID supaya konsisten dan tidak diubah setiap kali mengikuti selera programmer.

Reply:
Ini sy ikuti di style guide CI3

https://codeigniter.com/userguide3/general/styleguide.html#class-and-method-naming

INCORRECT:
```
if ($foo || $bar)
if ($foo AND $bar)  // okay but not recommended for common syntax highlighting applications
if (!$foo)
if (! is_array($foo))
```

CORRECT:
```
if ($foo OR $bar)
if ($foo && $bar) // recommended
if ( ! $foo)
if ( ! is_array($foo))
```

Misal klau ada perubahan nanti sy tambahkan di aturan penulisan.